src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -82,6 +83,25 @@
     long timerExpire;
     boolean attemptRefreshTokenLocked = false;
 
+    // Threasholds related to when prepared statement handles are cleaned-up. 1 == immediately.
+    /**
+     * The initial default on application start-up for the prepared statement clean-up action threshold (i.e. when sp_unprepare is called). 
+     */    
+    static final public int INITIAL_DEFAULT_SERVER_PREPARED_STATEMENT_DISCARD_THRESHOLD = 10; // Used to set the initial default, can be changed later.
+    static private int defaultServerPreparedStatementDiscardThreshold = -1; // Current default for new connections
+    private int serverPreparedStatementDiscardThreshold = -1; // Current limit for this particular connection.
+
+    /**
+     * The initial default on application start-up for if prepared statements should execute sp_executesql before following the prepare, unprepare pattern. 
+     */    
+    static final public boolean INITIAL_DEFAULT_ENABLE_PREPARE_ON_FIRST_PREPARED_STATEMENT_CALL = false; // Used to set the initial default, can be changed later. false == use sp_executesql -> sp_prepexec -> sp_execute -> batched -> sp_unprepare pattern, true == skip sp_executesql part of pattern.
+    static private Boolean defaultEnablePrepareOnFirstPreparedStatementCall = null; // Current default for new connections
+    private Boolean enablePrepareOnFirstPreparedStatementCall = null; // Current limit for this particular connection.
+
+    // Handle the actual queue of discarded prepared statements.
+    private ConcurrentLinkedQueue<PreparedStatementDiscardItem> discardedPreparedStatementHandles = new ConcurrentLinkedQueue<PreparedStatementDiscardItem>();
+    private AtomicInteger discardedPreparedStatementHandleQueueCount = new AtomicInteger(0);
```

Comment:
It seems like `discardedPreparedStatementHandleQueueCount ` is tracking the count of items in `discardedPreparedStatementHandles `. Is this right? If yes, can you use `discardedPreparedStatementHandles.size()` to get this count, instead of declaring a new variable?